src/app-layer-register.h

```diff
@@ -0,0 +1,91 @@
+/* Copyright (C) 2017 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Pierre Chifflier <chifflier@wzdftpd.net>
+ */
+
+#ifndef __APP_LAYER_REGISTER_H__
+#define __APP_LAYER_REGISTER_H__
+
+typedef struct AppLayerParser {
+    const char *name;
+    const char *default_port;
+    int ip_proto;
+    const char *proto_name;
```

Comment:
What is the purpose or proto_name? It doesn't appear to be used in the register functions.

Otherwise good. Was able to migrate the DNP3 parser to these new functions with success.

Reply:
The intended use is to provide a way to differentiate the name (used as an identifier) from the display name. But it's true that it's currently unused. We may want to keep it (for later use) or remove, as you prefer.

Reply:
IMO, remove until needed.

Reply:
I've taken this into my '4.1 next' branch I'm currently QAing. Removed the unused field.